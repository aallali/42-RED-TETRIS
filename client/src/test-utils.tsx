// test-utils.jsx
import React, { ReactChildren } from 'react'
import { render as rtlRender } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
// Import your own reducer
import playerReducer from "./reducers/player.reducer";
import opponentReducer from "./reducers/opponent.reducer"
import errorReducer from "./reducers/error.reducer"
import gameReducer from "./reducers/game.reducer"
import chatReducer from './reducers/chat.reducer';

function aallaliStore(preloadedState: any) {
	return configureStore({
		reducer: {
			player: playerReducer,
			opponents: opponentReducer,
			error: errorReducer,
			game: gameReducer,
			chat: chatReducer
		},
		preloadedState
	})
}
function render(
	ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
	{
		preloadedState,
		store = configureStore({
			reducer: {
				player: playerReducer,
				opponents: opponentReducer,
				error: errorReducer,
				game: gameReducer,
				chat: chatReducer
			},
			preloadedState
		}),
		...renderOptions
	}: any = {}
) {
	function Wrapper({ children }: { children: ReactChildren }) {
		return <Provider store={store}>{children}</Provider>

	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}


// re-export everything
export * from '@testing-library/react'
// override render method
export { render, aallaliStore }