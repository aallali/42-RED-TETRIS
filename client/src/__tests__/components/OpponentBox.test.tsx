import renderer from "react-test-renderer";
import Opponents from "../../components/OpponentBox"
import { Provider } from 'react-redux';

import { aallaliStore, render, screen } from '../../test-utils'


describe("_ Opponenets Component test", () => {
	test("opponents listing render test ", () => {
		const store = aallaliStore({ opponents: { players: [{ name: "botOpponent", lost: true, score: 100, rows: 3, level: 2, admin: false }] } })
		const tree = renderer.create(<Provider store={store}>
			< Opponents gameOver={true} />
		</Provider>).toJSON();
		expect(tree).toMatchSnapshot();
	});
	test("opponents listing render test2 ", async () => {
		render(
			< Opponents gameOver={false} />, {
			preloadedState:
			{
				opponents: {
					players: [{
						name: "testOpponent",
						lost: false,
						stage: [],
						rows: 55,
						score: 55,
						level: 55,
						admin: true
					}]
				}
			}
		}
		)

		expect(await screen.findByText(/★/)).toBeInTheDocument()

		expect(await screen.findByTestId("gameOverOverlay")).not.toHaveClass('resourceanimation')
		expect(await screen.findByText(/winner/i)).toBeInTheDocument()
	});


	test("opponents listing render test3 ", async () => {
		render(
			< Opponents gameOver={true} />, {
			preloadedState:
			{
				opponents: {
					players: [{
						name: "testOpponent",
						lost: true,
						stage: [],
						rows: 0,
						score: 0,
						level: 0,
						admin: true
					}]
				}
			}
		}
		)

		expect(await screen.findByText(/★/)).toBeInTheDocument()
	
		expect(await screen.findByTestId("gameOverOverlay")).toHaveClass('resourceanimation')
		expect(await screen.findByText(/Lost/i)).toBeInTheDocument()
	});
})
