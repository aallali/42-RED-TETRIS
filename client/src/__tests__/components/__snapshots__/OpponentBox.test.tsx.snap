// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_ Opponenets Component test opponents listing render test  1`] = `
<div
  className="flex gap-1 flex-wrap"
>
  <div
    className="w-auto p-0"
  >
    <div
      className="bg-white p-0 rounded-md shadow-lg text-center"
    >
      <h2
        className="text-md font-bold text-gray-700"
      >
        #
        botOpponent
      </h2>
      <div
        className="flex flex-row  justify-center items-center"
      />
      <span
        className="font-bold block text-left ml-2 p-0 text-xs"
      >
        level : (
        2
        ) 
      </span>
      <span
        className="font-bold block text-left ml-2 p-0 text-xs"
      >
        Score : (
        100
        ) 
      </span>
      <span
        className="font-bold block text-left ml-2 p-0 text-xs"
      >
        Rows c. : (
        3
        ) 
      </span>
    </div>
  </div>
</div>
`;
